community/kernel/src/main/java/org/neo4j/kernel/impl/store/format/RecordFormats.java

```diff
@@ -44,9 +48,15 @@ public Factory( String key, String... altKeys )
 
     String storeVersion();
 
+    /**
+     * @return format generation, with the intent of usage being that a store can migrate to a newer or
+     * same generation, but not to an older generation.
+     */
+    int generation();
```

Comment:
i find it hard to follow and understand why we need it in addition to versions of particular formats, especially when generation following completely independent from numbering scheme, it's really confusing